### ISSUE TITLE
Conflicts across class namespaces

### ISSUE THREAD

**Issue by dmazan** [ruby-grape/grape-swagger-entity#13]
When multiple namespaces contain the same entity names, grape-swagger-entity confuses them.  Consider the following example:

class Class1
  class Entity < Grape::Entity
    expose :one_thing
  end
end

class Class2
  class Entity < Grape::Entity
    expose :another_thing
  end
end

class Something < Grape::API
  add_swagger_documentation models: [Class1::Entity, Class2::Entity]
end

grape-swagger-entity incorrectly applies Class2::Entity to both models and refers to both a simply "Entity".  Change it to Class1::Entity1 and Class2::Entity2 and it works correctly.

Better, I think, if the gem understands the Class1::Entity and Class2::Entity are different things and uses their fully-qualified names in the resulting documentation.

**Comment by Bugagazavr** [comment]
Hi, thanks, this is not issue of grape-swagger-entitty, this issue is a part of grape-swagger https://github.com/ruby-grape/grape-swagger/blob/master/lib/grape-swagger/endpoint.rb#L291 all classes going to be demodulize

As workaround you can apply this patch when your application initialized 

```rb
require 'grape-swagger/endpoint'

module Grape
  class Endpoint
    def model_name(m)
      m.to_s.gsub('::', '')
    end
  end
end
```

**Comment by Bugagazavr** [comment]
cc @LeFnord

**Comment by LeFnord** [comment]
yeap it is a grape-swagger thing as @Bugagazavr said,
see here: https://github.com/ruby-grape/grape-swagger/issues/430#issuecomment-223449755

@dmazan 
> Better, I think, if the gem understands the Class1::Entity and Class2::Entity are different things and uses their fully-qualified names in the resulting documentation.

you mean this `Class1Entity` and `Class2Entity`, as definition names, right?

**Comment by LeFnord** [comment]
@dmazan … please have a look on the PR
the definition names are now without `Entity`

**Comment by Bugagazavr** [comment]
fixed in 0.24.0

**Comment by LeFnord** [comment]
@Bugagazavr … yeap, that's temawork 😉
